plugin-enclave-gradle/src/main/kotlin/com/r3/conclave/plugin/enclave/gradle/gramine/GenerateGramineBundle.kt

```diff
@@ -85,6 +89,61 @@ open class GenerateGramineBundle @Inject constructor(
         }
     }
 
+    private fun findJavaModules(): String {
+        val enclaveJar = outputDir.file(GRAMINE_ENCLAVE_JAR).get().asFile.absolutePath
+
+        val dependentModules = execCommand(
+            "jdeps",
+            "--print-module-deps",
+            "--ignore-missing-deps",
+            enclaveJar
+        ).run { trimEnd().replace(System.lineSeparator(), "").split(",") }
+        // TODO: Provide mechanism to allow the user to add modules needed when using reflection
+        val defaultModules = listOf("java.base", "java.desktop", "java.sql", "jdk.crypto.ec")
+        return (defaultModules + dependentModules).distinct().joinToString(separator = ",")
+    }
+
+    private fun createCustomJDK() {
+        val jlinkOutputPath = "${outputDir.get()}/$JLINK_OUTPUT_DIRECTORY"
```

Comment:
We should be creating a tar gz file containing the jdk image and the libs. It will preserve the permission flags on the files and enable better compression ability (since everything is compressed together in a tar file).

Reply:
You mean that a tar gz would compress more than what is already compressed by the bundle zip? Happy to do so, I just thought it was a bit weird to zip (with zip and tar cz) twice

Reply:
Yes, a tar gz will compress better and it will preserve the flags on the files.